From 39e4eda52783cc3a78ad5d50765d9a2a605e8217 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 30 Aug 2016 18:01:21 +0500
Subject: [PATCH] CMakeLists: add include path so Xlib.h is found as needed

All the targets including vk_platform.h or directly including
X11/Xlib.h require to know the directory for the installed
header. Add the directory to these so the requirements are
filled in properly.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 CMakeLists.txt                                                       | 2 ++
 layersvt/CMakeLists.txt                                              | 1 +
 vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/CMakeLists.txt | 1 +
 vktrace/src/vktrace_layer/CMakeLists.txt                             | 1 +
 vktrace/src/vktrace_replay/CMakeLists.txt                            | 1 +
 5 files changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 898b0f9..b99f1c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,6 +185,8 @@ find_program(GLSLANG_VALIDATOR NAMES glslangValidator
                    "${GLSLANG_BINARY_ROOT}/StandAlone"
                    "${PROJECT_SOURCE_DIR}/external/${BINDATA_DIR}")
 
+find_path(X11_XLIB_INCLUDE_DIR X11/Xlib.h DOC "Path to X11/Xlib.h")
+
 find_path(GLSLANG_SPIRV_INCLUDE_DIR SPIRV/spirv.hpp HINTS "${EXTERNAL_SOURCE_ROOT}/glslang"
                                                     "${CMAKE_SOURCE_DIR}/../glslang"
                                               DOC "Path to SPIRV/spirv.hpp")
diff --git a/layersvt/CMakeLists.txt b/layersvt/CMakeLists.txt
index 32ee1c4..8595418 100644
--- a/layersvt/CMakeLists.txt
+++ b/layersvt/CMakeLists.txt
@@ -95,6 +95,7 @@ include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/../include/vulkan
     ${CMAKE_CURRENT_BINARY_DIR}
     ${PROJECT_SOURCE_DIR}/../glslang/SPIRV
+    ${X11_XLIB_INCLUDE_DIR}
 )
 
 if (WIN32)
diff --git a/vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/CMakeLists.txt b/vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/CMakeLists.txt
index a62dc69..209ecf5 100644
--- a/vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/CMakeLists.txt
+++ b/vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/CMakeLists.txt
@@ -52,6 +52,7 @@ include_directories(
     ${CODEGEN_VKTRACE_DIR}
     ${VKTRACE_VULKAN_INCLUDE_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}/../vulkan/codegen_utils
+    ${X11_XLIB_INCLUDE_DIR}
 )
 # needed for vktraceviewer_vk library which is shared
 if (NOT MSVC)
diff --git a/vktrace/src/vktrace_layer/CMakeLists.txt b/vktrace/src/vktrace_layer/CMakeLists.txt
index 88e15e8..864ec20 100644
--- a/vktrace/src/vktrace_layer/CMakeLists.txt
+++ b/vktrace/src/vktrace_layer/CMakeLists.txt
@@ -82,6 +82,7 @@ include_directories(
     ${VKTRACE_VULKAN_INCLUDE_DIR}
     ${CODEGEN_UTILS_DIR}
     ${CMAKE_CURRENT_BINARY_DIR}/../../../layersvt
+    ${X11_XLIB_INCLUDE_DIR}
 )
 # copy/link layer json file into build/layersvt directory
 if (NOT WIN32)
diff --git a/vktrace/src/vktrace_replay/CMakeLists.txt b/vktrace/src/vktrace_replay/CMakeLists.txt
index 5bc1807..4fdeb19 100644
--- a/vktrace/src/vktrace_replay/CMakeLists.txt
+++ b/vktrace/src/vktrace_replay/CMakeLists.txt
@@ -18,6 +18,7 @@ include_directories(
     ${SRC_DIR}/vktrace_common
     ${SRC_DIR}/thirdparty
     ${CMAKE_CURRENT_SOURCE_DIR}/../vktrace_extensions/vktracevulkan/vkreplay/
+    ${X11_XLIB_INCLUDE_DIR}
 )
 
 set (LIBRARIES vktrace_common vulkan_replay)
diff --git a/vktrace/src/vktrace_common/CMakeLists.txt b/vktrace/src/vktrace_common/CMakeLists.txt
index 07ffba7..8fcbb39 100644
--- a/vktrace/src/vktrace_common/CMakeLists.txt
+++ b/vktrace/src/vktrace_common/CMakeLists.txt
@@ -6,6 +6,7 @@ include(${SRC_DIR}/build_options.cmake)
 include_directories(
     ${SRC_DIR}/vktrace_common
     ${SRC_DIR}/thirdparty
+    ${X11_XLIB_INCLUDE_DIR}
 )
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
diff --git a/vktrace/src/vktrace_trace/CMakeLists.txt b/vktrace/src/vktrace_trace/CMakeLists.txt
index 63ba576..b1c56b3 100644
--- a/vktrace/src/vktrace_trace/CMakeLists.txt
+++ b/vktrace/src/vktrace_trace/CMakeLists.txt
@@ -16,6 +16,7 @@ include_directories(
     ${SRC_DIR}
     ${SRC_DIR}/vktrace_common
     ${SRC_DIR}/vktrace_trace
+    ${X11_XLIB_INCLUDE_DIR}
 )
 
 add_executable(${PROJECT_NAME} ${SRC_LIST})
-- 
1.9.1

